File: DebriefRepository/src/main/java/org/example/debriefrepository/entity/Group.java
package org.example.debriefrepository.entity;

import jakarta.persistence.*;
import jakarta.persistence.Column;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "group", schema = "debrief_mgmt")
public class Group extends BaseEntity {

    @Column(name = "name", nullable = false)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "commander_id")
    private User commander;

    @OneToMany(mappedBy = "group")
    private List<Debrief> debriefs = new ArrayList<>();

    @OneToMany(mappedBy = "group")
    private List<User> users = new ArrayList<>();

}

File: DebriefRepository/src/main/java/org/example/debriefrepository/entities/Group.kt
package org.example.debriefrepository.entities

import jakarta.persistence.*

@Entity
@Table(name = "\"group\"", schema = "debrief_mgmt")
open class Group {
    @Id
    @Column(name = "id", nullable = false)
    open var id: Long? = null

    @Column(name = "name", nullable = false, length = Integer.MAX_VALUE)
    open var name: String? = null

    @ManyToMany(mappedBy = "groups")
    open var debriefs: MutableSet<Debrief> = mutableSetOf()

    @ManyToMany(mappedBy = "groups")
    open var users: MutableSet<org.example.debriefrepository.entities.User> = mutableSetOf()
}